eds_scikit/utils/framework.py

```diff
@@ -60,21 +65,142 @@ def pandas(obj: DataObject) -> DataObject:
 
 
 def koalas(obj: DataObject) -> DataObject:
-    if get_framework(obj) is _koalas:
+    if get_framework(obj) is ks:
         return obj
     try:
         return obj.to_koalas()
     except AttributeError:
         pass
 
     # will raise ValueError if impossible
-    return _koalas.from_pandas(obj)
-
-
-def add_unique_id(obj: DataFrame, col_name: str = "id") -> DataFrame:
-    fw = get_framework(obj)
-    if fw == _pandas:
-        obj[col_name] = range(len(obj))
-        return obj
-    else:
-        return obj.koalas.attach_id_column(id_type="distributed", column=col_name)
+    return ks.from_pandas(obj)
+
+
+class BackendDispatcher:
+    """Dispatcher between pandas, koalas and custom method."""
+
+    def get_backend(self, obj: DataObject) -> Optional[ModuleType]:
+        for backend in VALID_FRAMEWORKS:
+            if obj.__class__.__module__.startswith(backend.__name__):
+                return backend
+        return None
+
+    def is_pandas(self, obj: DataObject) -> bool:
+        return self.get_backend(obj) is pd
+
+    def is_koalas(self, obj: DataObject) -> bool:
+        return self.get_backend(obj) is ks
+
+    def to_pandas(self, obj: DataObject) -> DataObject:
+        if self.get_backend(obj) is pd:
+            return obj
+        try:
+            return obj.to_pandas()
+        except AttributeError:
+            raise ValueError("Could not convert object to pandas.")
+
+    def to_koalas(self, obj: DataObject) -> DataObject:
+        if self.get_backend(obj) is ks:
+            return obj
+        try:
+            return obj.to_koalas()
+        except AttributeError:
+            pass
+        # Will raise ValueError if impossible
+        return ks.from_pandas(obj)
```

Comment:
Two suggestions on this:
- It might be useful to also expose a simple `to` method, with a `backend` parameter
- Also handling iterable or dictionaries of `DataObject` can be a plus.

Those changes would especially make testing (with the appropriate fixture) way cleaner and concise !

Reply:
1. I'm a bit hesitant on this one, don't you think it will introduce "yet another method of converting" since we already have `to_pandas` and `to_koalas`?
2. Good idea, will do

Reply:
In my opinion it would go in pair with the `backend` kwarg that we're already exposing when calling a method (e.g. when doing `bd.cut(..., backend="pd", ...)`,  and it could be useful to dynamically pick between `to_pandas` and `to_koalas` (other than via `__getattr__`). But we can add it later if we really need it !

Reply:
Ok let's add it!